mux.go

```diff
@@ -112,7 +112,7 @@ func Register[T, B any](s *Server, route Route[T, B], controller http.Handler, o
 	route.Path = s.basePath + route.Path
 
 	var err error
-	route.Operation, err = RegisterOpenAPIOperation(s, route)
+	route.Operation, err = RegisterOpenAPIOperation(s.OpenAPI, route)
```

Comment:
If you are passing route, to set a field of route

Why is it the signature like this?


```suggestion
	err = route.RegisterOpenAPIOperation(s.OpenAPI)
```

I know I'm suggesting a code change, but it can be retro compatible

Reply:
Ob maybe because you are relying g on the magic of generics